longtask-timing/idlharness.window.js

```diff
@@ -0,0 +1,46 @@
+// META: global=window,worker
+// META: script=/resources/WebIDLParser.js
+// META: script=/resources/idlharness.js
+
+// https://w3c.github.io/longtask-timing/
+
+'use strict';
+
+promise_test(async t => {
+  const srcs = ['longtasks', 'performance-timeline'];
+  const [idl, perf] = await Promise.all(
+    srcs.map(i => fetch(`/interfaces/${i}.idl`).then(r => r.text())));
+
+  const idl_array = new IdlArray();
+  idl_array.add_idls(idl);
+  idl_array.add_dependency_idls(perf);
+
+  const testIdls = new Promise(resolve => {
+    const observer = new PerformanceObserver(entryList => {
+      const entries = Array.from(entryList.getEntries());
+      const attribution = entries
+          .reduce((sum, e) => sum.concat(e.attribution || []), []);
+      idl_array.add_objects({
+        PerformanceLongTaskTiming: entries,
+        TaskAttributionTiming: attribution,
+      });
+      idl_array.test();
+      resolve();
+    });
+    observer.observe({entryTypes: ['longtask']});
+  });
+
+  const taskTime = 100;
+  const longTask = () => {
+    var begin = self.performance.now();
+    while (self.performance.now() < begin + taskTime);
+  }
+  t.step_timeout(longTask, 0);
+
+  const timeout = new Promise(
+      (_, reject) => t.step_timeout(reject, 2 * taskTime));
```

Comment:
Is this needed? This seems to assume that the part of the test that does not involve the long task takes less than 100ms.

Reply:
It's supposed to just be a fast-fail really. Yeah, it does assume that, though `step_timeout` is enforced so that the settings scale it for slower environments:

> The difference between setTimeout and step_timeout is that the latter takes account of the timeout multiplier when computing the delay; e.g., in the above case a timeout multiplier of 2 would cause a pause of 4000ms before calling the callback. This makes it less likely to produce unstable results in slow configurations.

https://web-platform-tests.org/writing-tests/testharness-api.html#timeouts-in-tests

That said, this could be a case of "works on my box" at its finest.. 50ms is the threshold for a "long" task. Doubled it and .. doubled that. lol. Want me to bump it up to a higher value?

Reply:
The assumption that a 100ms task is considered a longtask seems fine. The assumption that a valid test run will take less than 200ms less so. I'd prefer some simple check to fail right away when longtasks are not supported at all. With this, not sure this early rejection is needed. Anyways, this is still better than the other longtask tests that look like they'll just Timeout in unsupported browsers. So I'm ok with an increase in the rejection time, which is currently 200 ms (or just start the rejection clock after executing the long task).

Reply:
kk - In https://github.com/web-platform-tests/wpt/pull/11815/commits/72262b4f8bb72a5353fda441ada7124492167b4b I've try-catched the object creation, changed the timeout to 1000ms, and run the (partial) IdlArray.test on rejection. 

In FF, it's 23 tests (18 fails, 5 passes) and Chrome it's 32 (all passes).